### PR TITLE
Working LoginChat module

### PR DESCRIPTION
**Pull request by ghost**
I have been able to build it on docker. Can you see if this resolves your issue?

### CODE REVIEW

File: src/LoginChat.cpp [hunk 1]
```diff
@@ -14,48 +14,43 @@ class LoginChat : public PlayerScript{
...
             sConfigMgr->LoadMore(cfg_file.c_str());
         }
     }
 };
 
-
 void AddLoginChatScripts() {
-    new LoginChat();
-    new login_chat_conf();
+    new LoginChat;
```
Comment (Nefertumm) [inline src/LoginChat.cpp:54]: Missing ()

File: src/LoginChat.cpp [hunk 2]
```diff
@@ -14,48 +14,43 @@ class LoginChat : public PlayerScript{
...
+    new LoginChat_Config;
```
Comment (Nefertumm) [inline src/LoginChat.cpp:55]: Here too

File: src/LoginChat.cpp [hunk 3]
```diff
@@ -14,48 +14,43 @@ class LoginChat : public PlayerScript{
...
+          {
+            if (Channel* channel = cMgr->GetJoinChannel(channelName, channelId))
+              channel->JoinChannel(player, password);
+          }
+        }
     }
 };
 
-class login_chat_conf : public WorldScript
+class LoginChat_Config : public WorldScript
```
Comment (Winfidonarleyan) [inline src/LoginChat.cpp:35]: Useless func

**Comment by disclosurez** [comment]
will test and let you know thanks!

**Comment by disclosurez** [comment]
player doesnt auto join the channel?
perhaps i did something wrong?

the channel name exists in this database>

**Comment by disclosurez** [comment]
change this:
`
          QueryResult result = CharacterDatabase.PQuery("SELECT channelId password FROM Channels WHERE name = '%s'", channelName.c_str());`

to this:

`          QueryResult result = CharacterDatabase.PQuery("SELECT channelId password FROM channels WHERE name = '%s'", channelName.c_str());`

**Comment by disclosurez** [comment]
doesn't work at all, causes crash when player logs in.

File: src/LoginChat.cpp [hunk 4]
```diff
@@ -14,48 +14,43 @@ class LoginChat : public PlayerScript{
...
-    	{
-        	if (Channel* channel = cMgr->GetJoinChannel("world", 9))
-            		channel->JoinChannel(player, "");
-    		cMgr->LoadChannels();
-	}
-
+        if (sConfigMgr->GetBoolDefault("LoginChat.enabled", true))
+        {
+          std::string channelName = sConfigMgr->GetStringDefault("LoginChat.name", "world");
+          QueryResult result = CharacterDatabase.PQuery("SELECT channelId password FROM Channels WHERE name = '%s'", channelName.c_str());
```
Comment (disclosurez) [inline src/LoginChat.cpp]:
```suggestion
          QueryResult result = CharacterDatabase.PQuery("SELECT channelId password FROM channels WHERE name = '%s'", channelName.c_str());
```

Comment (disclosurez) [reply src/LoginChat.cpp]: to fix build

**Review by disclosurez — APPROVED** [review]

**Comment by disclosurez** [comment]
doesn't work at all, causes crash when player logs in.

error:
`/azerothcore-wotlk/src/common/Database/QueryResult.h:36 in operator[] ASSERTION FAILED:
  index < _fieldCount

Segmentation fault (core dumped)
`

**Comment by disclosurez** [comment]
doesn't work at all, causes crash when player logs in.

error:
`/azerothcore-wotlk/src/common/Database/QueryResult.h:36 in operator[] ASSERTION FAILED:
  index < _fieldCount

Segmentation fault (core dumped)
`

File: src/LoginChat.cpp (hunk 4 above)
Comment (disclosurez) [inline src/LoginChat.cpp:20]:
```suggestion
          QueryResult result = CharacterDatabase.PQuery("SELECT channelId, password FROM channels WHERE name = '%s'", channelName.c_str());
```

Comment (disclosurez) [reply src/LoginChat.cpp:20]: fix build & crash

**Comment by disclosurez** [comment]
After above fixes, players are still not joined to channel on login.

perhaps this helps?

https://github.com/mynew3/crus-bud/blob/51134ded5224b857a3408819d2cfce3c11d0f3f8/src/server/scripts/Custom/Player/player_channel_pvp.cpp

**Pull request closed without merging**